lib/Crypto/DecryptAll.php

```diff
@@ -90,55 +100,104 @@ public function __construct(
 	 * @throws LoginException
 	 */
 	public function prepare(InputInterface $input, OutputInterface $output, $user) {
-
-		$question = new Question('Please enter the recovery key password: ');
-
+		// We need to setup the module with everything it needs in order to decrypt user data
+		// This could be using master key or user key encryption
+		// and could also be for a single user or multiple users
+
+		/**
+		 * Try to stop decrypting process if password method is used for decrypting all users.
+		 * If the decrypt-all command doesn't provide a user, which means decryption of oC
+		 * filesystem. And if the method provided is password then it should stop decryption
+		 * process, because same password can not be used for every user. Even though the
+		 * input user is empty the $user can have value of each user(s) in oC.
+		 */
+		if (($input->getArgument('user') === '') && ($input->getOption('method') === 'password')) {
```

Comment:
please use `empty($user)` here like it's done further below.

then also update the PHPDoc of this method to have the proper types for the "$user" argument as it's not clear whether it's a string or not. also it's not clear that it can be null and that null means "all"

Reply:
There is a small problem I see here to use `empty($user)` to satisfy the condition what I am trying to achieve. The `$user` is not empty considering the code path triggered from `encryption:decrypt-all` command. 

Let say user-keys encryption is enabled and admin wants to run decrypt-all on all users. And the command triggered was:
`./occ encryption:decrypt-all -m password -c yes` 
This would reach the line
- https://github.com/owncloud/core/blob/master/lib/private/Encryption/DecryptAll.php#L176
- which would eventually call `prepare()` and thus prepare will have uid of the user passed. So that way `$user` the uid cannot be empty :(

The place where `$user` becomes `""` is while decrypting masterkey encrypted files.

Reply:
so does it mean that the check for `empty($user)` on line 137 is never called either ?

Reply:
ok, please update the PHPDoc of the method to say so: when is $user an empty string and what does it mean.
this place is already confusing enough

Reply:
I have updated the PHPDoc of the method.